[VAULT-46135] Add docs for Terraform Secret Engine Root rotation (#2797)
hashicorp/web-unified-docs: content/vault/v2.x/content/docs/secrets/terraform.mdx

- For **organization tokens**, Terraform implicitly replaces the token when Vault
  creates a new one, so Vault performs no explicit revocation.
- For **team and user tokens**, Vault revokes the previous token after storing the
  new one, as long as it knows the token ID.

Vault determines the ID of the current token from the `auth-token` link in the
[`/account/details`](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/account#sample-response)
response. HCP Terraform and Terraform Enterprise v2.1 and later return this link,
so Vault revokes the previous team or user token on every rotation.

Earlier versions of Terraform Enterprise do not return the `auth-token` link, so
Vault cannot determine the ID of the token you configured manually. The first
time Vault rotates that token, it returns a warning asking you to revoke the
initial token manually in Terraform. Vault records the ID of each token it
creates, so subsequent rotations revoke the previous token automatically.

~> **Note:** On-demand rotation with `terraform/rotate-root` is available in both
Vault Community and Enterprise editions. Scheduled, automatic rotation requires

> Vault Enterprise.

Reviewer: Let's use the callout box syntax suggested by the [style guide](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/appendix.md#alert-types).

```suggestion
<Note>

On-demand rotation with `terraform/rotate-root` is available in both
Vault Community and Enterprise editions. Scheduled, automatic rotation requires
Vault Enterprise.

</Note>
```

token           <example token>
token_id        at-fqvtdTQ5kQWcjUfG
```

## Root token rotation

The root token configured at `terraform/config` is the token Vault uses to

> authenticate with HCP Terraform or Terraform Enterprise. Vault can rotate this

Reviewer 2: ```suggestion
authenticate with HCP Terraform or Terraform Enterprise. Vault can rotate the root token
```

token so that only Vault knows its value.

Trigger an on-demand rotation with the `terraform/rotate-root` endpoint:

```shell-session
$ vault write -f terraform/rotate-root
Key                    Value
---                    -----
last_vault_rotation    2026-04-02T23:30:00Z
owner_id               user-glhf1234
token_owner_type       user
```

Vault creates a new token of the same type as the configured token, validates
it, stores it as the new root token, and revokes the previous token. The
behavior differs slightly by token type:

- For **organization tokens**, Terraform implicitly replaces the token when Vault
  creates a new one, so Vault performs no explicit revocation.

> - For **team and user tokens**, Vault revokes the previous token after storing the
>   new one, as long as it knows the token ID.

Reviewer 2: ```suggestion
- For **team and user tokens**, Vault revokes the previous token after storing the
  new one, as long Vault can determine the previous token ID.
```
Style correction: avoid pronouns when possible

> Vault determines the ID of the current token from the `auth-token` link in the
> [`/account/details`](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/account#sample-response)
> response. HCP Terraform and Terraform Enterprise v2.1 and later return this link,
> so Vault revokes the previous team or user token on every rotation.

Reviewer 2: ```suggestion
Vault revokes the previous team or user token on every rotation when it can
determine the ID of the current token from the `auth-token` link in the
[`/account/details`](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/account#sample-response)
response returned by HCP Terraform and Terraform Enterprise v2.1+.
```
We can simplify the sentences here a bit

Earlier versions of Terraform Enterprise do not return the `auth-token` link, so
Vault cannot determine the ID of the token you configured manually. The first
time Vault rotates that token, it returns a warning asking you to revoke the
initial token manually in Terraform. Vault records the ID of each token it
creates, so subsequent rotations revoke the previous token automatically.

> <Note>

Reviewer 2: Is this note really needed?
